### DISCUSSION TITLE
chainWebpack：请问如何自定义Loader, 代码如下：

### DISCUSSION THREAD

**Discussion by ganquan0910** [umijs/umi#8218] [Q&A] [answered]
export default {
    npmClient: "pnpm",
    plugins: [require.resolve("@umijs/plugins/dist/model"), require.resolve("@umijs/plugins/dist/antd")],
    antd: {},
    model: {},
    history: { type: "hash" },
    publicPath: process.env.RUN_ENV === "build" ? "./" : "/",
    chainWebpack(config, {}) {
        config.module
            .rule("graphql")
            .test(/\.(graphql|gql)$/)
            .use("graphql-tag")
            .loader("graphql-tag/loader");
    },
};

import getApplicationInfoByAiUuid from "./getApplicationInfoByAiUuid.gql";

console.log(getApplicationInfoByAiUuid);

输出：

/static/getApplicationInfoByAiUuid.faefa1c0..gql

没有加载成功，请问是什么问题？

**Comment by fz6m** [comment, marked as answer]
```bash
  pnpm add -D graphql-tag graphql
```

```ts
// .umirc.ts

export default {
  chainWebpack(config) {

    const GQL_REG = /\.(graphql|gql)$/

    config.module.rule('asset').exclude.add(GQL_REG).end();

    config.module
      .rule('graphql')
      .test(GQL_REG)
      .exclude.add(/node_modules/)
      .end()
      .use('graphql-tag')
      .loader('graphql-tag/loader')
      .end()

  },
};

```

**Reply by ganquan0910** [reply]
@fz6m 非常感谢！
